xmodule/tests/test_library_content.py

```diff
@@ -53,13 +57,25 @@ def setUp(self):
             max_count=1,
             source_library_id=str(self.library.location.library_key)
         )
+        self.lc_block.runtime._services.update({'library_tools': self.tools})  # pylint: disable=protected-access
+
+    def _sync_lc_block_from_library(self, upgrade_to_latest=False):
+        """
+        Save the lc_block, then sync its children with the library, and then re-load it.
+
+        We must re-load it because the syncing happens in a Celery task, so that original self.lc_block instance will
+        not have changes manifested on it, but the re-loaded instance will.
+        """
+        self.store.update_item(self.lc_block, self.user_id)
+        self.lc_block.sync_from_library(upgrade_to_latest=upgrade_to_latest)
+        self.lc_block = self.store.get_item(self.lc_block.location)
```

Comment:
the save/sync/reload pattern was need in a ton of test cases, so I pulled up to here.